src/qibo/hamiltonians/hamiltonians.py

```diff
@@ -566,8 +568,8 @@ def expectation_from_samples(self, freq, qubit_map=None):
                 if subk.count(1) % 2 == 1:
                     expval_k = -1
                 expval_q += expval_k * counts[i]
-            expval += expval_q * float(coeff[j])
-        return expval
+            expval += expval_q * self.terms[j].coefficient.real
```

Comment:
Any clue why `SymbolicTerm` coefficient are stored as complex values?
https://github.com/qiboteam/qibo/blob/c3031c606464650856d6d278b799eeee4a1f49fe/src/qibo/hamiltonians/terms.py#L138

@andrea-pasquale?

Reply:
I think `terms` here may not necessarily be hermitian. A term is a product of symbols and it happens that the symbols we provide are the Paulis which are all hermitian, however the user can also define their own symbol using custom matrix. So there may be a case where you have non-hermitian terms and complex coefficients resulting to a hermitian Hamiltonian.

The constant, on the other hand, probably needs to be real in every case, since it is multiplying the idenity that is hermitian.